pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -1185,3 +1188,105 @@ def test_config_repo_mirror_sync(
         )
         assert bytes("gpgcheck=1\n", "utf-8") in content
         assert bytes("repo_gpgcheck=0", "utf-8") in content
+
+
+@pytest.fixture
+def repo_4073_url(
+    tmp_path,
+    rpm_repository_factory,
+    rpm_rpmremote_factory,
+    rpm_publication_factory,
+    rpm_distribution_factory,
+    init_and_sync,
+):
+    REPODIR = tmp_path / "repo_4073"
+
+    def download_fixture(fixture_name: str, to_relative_path: str) -> Path:
+        url = urljoin(RPM_UNSIGNED_FIXTURE_URL, fixture_name)
+        response = requests.get(url)
+        response.raise_for_status()
+        pkg = REPODIR / to_relative_path
+        pkg.parent.mkdir(parents=True, exist_ok=True)
+        pkg.write_bytes(response.content)
+        return pkg
+
+    def get_package_list_from_repodata(repodata_dir: Path):
+        assert repodata_dir.exists()
+        primary_xml_file = next(repodata_dir.glob("*primary.xml*"))
+        result_repo = cr.RepositoryReader.from_metadata_files(str(primary_xml_file), None, None)
+        parsed = result_repo.parse_packages(only_primary=True)[0]  # {<checksum>: <cr.Package>}
+        return list(parsed.values())
+
+    # Setup repository packages layout
+    BEAR_FILENAME = "bear-4.1-1.noarch.rpm"
+    BEAR_RELATIVE_PATH = f"bear/{BEAR_FILENAME}"
+    CAMEL_FILENAME = "camel-0.1-1.noarch.rpm"
+    CAMEL_RELATIVE_PATH = f"camel/{BEAR_FILENAME}"  # yes, we're faking the filename
+
+    bear_rpm = download_fixture(BEAR_FILENAME, to_relative_path=BEAR_RELATIVE_PATH)
+    camel_rpm = download_fixture(CAMEL_FILENAME, to_relative_path=CAMEL_RELATIVE_PATH)
+    bear_pkg = cr.package_from_rpm(str(bear_rpm), location_href=BEAR_RELATIVE_PATH)
+    camel_pkg = cr.package_from_rpm(str(camel_rpm), location_href=CAMEL_RELATIVE_PATH)
+    BEAR_NAME = bear_pkg.name
+    CAMEL_NAME = camel_pkg.name
+
+    # Create repository with messed location_hrefs
+    NO_COMPRESSION = 2
+    with cr.RepositoryWriter(str(REPODIR), compression=NO_COMPRESSION) as writer:
```

Comment:
Just use the constant `cr.NO_COMPRESSION`

Reply:
This is an excellent usage of createrepo_c, btw.  Exactly what I was hoping would be made easy/possible with the new API.  Great job!